bin/collator/src/parachain/service.rs

```diff
@@ -606,6 +602,12 @@ pub struct StartupConfiguration {
     pub proposer_soft_deadline_percent: u8,
 }
 
+/// To add additional config to start_xyz_node functions
+pub struct AdditionalConfig {
```

Comment:
I understand that this might be a bit confusing - the item you are working on was created before the above struct, `StartupConfiguration` was added.

My suggestion was to add something like this `AdditionalConfig` instead of just adding new params like `evm_tracing_config` and `enable_evm_rpc`. If you check the git history of this file, it will be more clear.

Imo, we don't need both `StartupConfiguration` and `AdditionalConfig`.
And if we do, I don't think they should be _separate_ - `AdditionalConfig` should be a filed in `StartupConfiguration`.

Reply:
Any comment about this one? :slightly_smiling_face: 
Does having `StartupConfig` and `AdditionalConfig` make sense?

As mentioned above, when the ticket was created, `StartupConfig` didn't exist.
So having `AdditionalConfig` seems redundant (to me).

Imo, having something like this would be the best approach:

```
pub async fn start_astar_node(
     parachain_config: Configuration,
     polkadot_config: Configuration,
     collator_options: CollatorOptions,
     id: ParaId,
    additional_config: AdditionalConfig,
) -> sc_service::error::Result<(
    TaskManager,
    Arc<TFullClient<Block, astar::RuntimeApi, NativeElseWasmExecutor<astar::Executor>>>,
)> {

struct AdditionalConfig {
    evm_tracing_config: EvmTracingConfig,
    enable_evm_rpc: bool,
    // and here we can add anything else we need in the future
}
```

Reply:
I was not able to understand what exactly you meant so I thought I would discuss this later. Thanks for more explanation.
To me, it is a matter of the amount of abstraction that you we want. I don't mind having it either way but let's do it your proposed way.

Reply:
They should be part of additional config, imo.

I'd say - if it's not part of existing param structs, and is specific to us (like these two params), put it under additional config.